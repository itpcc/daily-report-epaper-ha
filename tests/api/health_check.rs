use axum::http::StatusCode;
use insta::assert_debug_snapshot;
use serde_json::Value;

use crate::helpers::*;

#[tokio::test]
async fn health_check_ok() {
    let app = TestApp::new().await;

    let resp = app.get("/health_check").await;
    let headers = resp.headers().clone();

    assert_eq!(resp.status(), StatusCode::OK);
    assert_debug_snapshot!(resp.json::<Value>().await.unwrap());
    assert!(headers.get("x-request-id").is_some());
    assert_eq!(headers.get("access-control-allow-origin").unwrap(), "*");
    assert!(headers.get("vary").is_some());
}

#[tokio::test]
async fn db_connection_ok() {
    let app = TestApp::new().await;
    assert_eq!(app.db.size(), 1);
}
