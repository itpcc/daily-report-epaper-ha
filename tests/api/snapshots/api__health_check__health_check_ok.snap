---
source: tests/api/health_check.rs
expression: "resp.json::<Value>().await.unwrap()"
---
Object {
    "status": String("ok"),
}
